tests/lib/pkgdb.sh

```diff
@@ -516,3 +605,18 @@ install_pkg_dependencies(){
     # shellcheck disable=SC2086
     distro_install_package $pkgs
 }
+
+# upgrade distribution and indicate if reboot is needed by outputting 'reboot'
+# to stdout
+distro_upgrade() {
+    case "$SPREAD_SYSTEM" in
+        arch-*)
+            if pacman -Syu --noconfirm 2>&1 | grep -q "there is nothing to do" ; then
+                echo "reboot"
```

Comment:
🙄

Reply:
I know it's kind of odd. With pacman there's a problem with how dependcies are handled, i.e. they are not really versioned in the same way as rpm/dnf/yum/apt/dpkg do it, in other words partial upgrades are not supported. On top of this, the image we use may have been built some time ago, so we need to sync (`-y`) at some point. In order to be able to just `pacman -S` when installing the packages and for things to work I figured it's less trouble to just upgrade the running image when preparing project. This comes with a downside, when upgrading you cannot easily tell if you need to reboot or not (which you may in case the kernel got updated, or systemd). So instead of second guessing, I just opted to reboot the system.

In case there is nothing to upgrade pacman outputs:
```
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
:: Starting full system upgrade...
 there is nothing to do   <-- our needle
```

Reply:
Can you add that description as a comment please.